pytorch_lightning/utilities/auto_restart.py

```diff
@@ -72,6 +76,10 @@ def __iter__(self) -> Iterator[Any]:
             i += 1
 
         # here: i == self._current_iteration
+        if self._cached_state_dict is not None:
+            self._cached_state_dict = None
```

Comment:
```suggestion
```
You are setting `self._cached_state_dict = None` in line 91 as well.

Reply:
true, but will be needed later when adding the random state stuff.
I will re-introduce it in the follow up PR. thanks